### PR TITLE
Preserve blob tags should not fail  the transfer if SAS does not have enough permissions

### PR DESCRIPTION
**Pull request by pranavmalik-msft**
### Description
This pull request addresses the need for clearer error messages during transfer failures, specifically when the preserve-blob-tags option is enabled. Previously, the error response incorrectly indicated a 403 permission issue, as evidenced below:
image (6)

 This improvement will enable Storage Explorer to accurately inform users of transfer failures related to tag preservation, such as "Failed to set tags due to permissions error."
The new flow would return a better error message as can be observed below:
image

### CODE REVIEW

File: cmd/copy.go [hunk 1]
```diff
@@ -1652,6 +1652,9 @@ func (cca *CookedCopyCmdArgs) processCopyJobPartOrders() (err error) {
 	if err != nil {
 		if err == NothingToRemoveError || err == NothingScheduledError {
 			return err // don't wrap it with anything that uses the word "error"
+		} else if cca.S2sPreserveBlobTags {
```
Comment (siminsavani-msft) [inline cmd/copy.go:1655]: Shouldn't we check if the preserve blob tags flag is enabled **and** the error message is the AuthorizationPermissionMismatch?

**Comment by siminsavani-msft** [comment]
Personally, I would keep the service error along with the newly added message: "Failed to set blob tags due to permission error"

File: cmd/copy.go (hunk 1 above)
Comment (pranavmalik-msft) [reply cmd/copy.go:1655]: We are doing exactly the same thing here in line no 1655

**Review by siminsavani-msft — APPROVED** [review]

**Review by adreed-msft — APPROVED** [review]

File: cmd/copy.go [hunk 2]
```diff
@@ -1652,6 +1652,9 @@ func (cca *CookedCopyCmdArgs) processCopyJobPartOrders() (err error) {
 	if err != nil {
 		if err == NothingToRemoveError || err == NothingScheduledError {
 			return err // don't wrap it with anything that uses the word "error"
+		} else if cca.S2sPreserveBlobTags && strings.Contains(err.Error(), "AuthorizationPermissionMismatch") {
+			// If S2sPreserveBlobTags is true, return a more descriptive error message indicating permission issue.
+			return fmt.Errorf("Failed to set blob tags due to permission error: %s.\n", err)
```
Comment (adreed-msft) [inline cmd/copy.go:1657]: Is this the only way we can get this back? Could this not occur later in a job, after enumeration?

File: e2etest/zt_newe2e_basic_functionality_test.go [hunk 3]
```diff
@@ -160,4 +162,48 @@ func (s *BasicFunctionalitySuite) Scenario_Copy_EmptySASErrorCodes(svm *Scenario
...
+					},
+				}),
+			},
+			Flags: CopyFlags{
+				CopySyncCommonFlags: CopySyncCommonFlags{
+					Recursive:           pointerTo(true),
+					S2SPreserveBlobTags: pointerTo(true),
+				},
+			},
+			ShouldFail: false,
```
[the remark this answers is not in the document]
Comment (pranavmalik-msft) [reply e2etest/zt_newe2e_basic_functionality_test.go]: yes, transfer should fail as done in the latest commit.

File: cmd/copy.go (hunk 2 above)
Comment (pranavmalik-msft) [reply cmd/copy.go:1657]: this is the only place we can  get this back as of now , even after job enumeration we are coming here.

**Review by gapra-msft — CHANGES_REQUESTED** [review]
Please also make sure that this error message is returned when the source/destination is pointing to the blob level URL

**Comment by gapra-msft** [comment]
@pranavmalik-msft I don't see the addition of a test that checks that this behavior works when pointing to the blob level URL, which was my previous comment. 

The existing test checks this behavior when pointed to a container level URL.

**Review by adreed-msft — CHANGES_REQUESTED** [review]

**Pull request closed without merging**